src/hooks/useCurrentBlockTimestamp.ts

```diff
@@ -1,10 +1,20 @@
 import { BigNumber } from '@ethersproject/bignumber'
 import { useSingleCallResult } from 'lib/hooks/multicall'
+import { useEffect, useState } from 'react'
 
 import { useInterfaceMulticall } from './useContract'
 
 // gets the current timestamp from the blockchain
 export default function useCurrentBlockTimestamp(): BigNumber | undefined {
```

Comment:
would it be simpler to make `block` a primitive `Number` instead ?

Reply:
It would require converting it back to `BigNumber` for some uses, so I don't think it simplifies things overall.

Reply:
i don't think it needs to be a BigNumber at all though if i look at usages 🤷